### PR TITLE
[#54671526] Makes objects instances (by vertex) as real objects

### PR DESCRIPTION
**Pull request by AurL**
When the DupliVerts tool is used in Blender on an object, its children are
duplicated according to each of its vertexes (i.e an instance of the children is
created and 'snapped' on each of the vertexes of the parent).
We now create a real object for each instance in order to retrieve these geometries.
The 'duplicates_make_real()' allows to do this.

### CODE REVIEW

File: blender-2.5/exporter/osg/osgdata.py [hunk 1]
```diff
@@ -157,6 +157,27 @@ def findArmatureObjectForTrack(track):
...
 
 
+def unselect_all_objects():
+    for obj in bpy.context.selected_objects:
+        obj.select = False
+
+
+def make_dupliverts_real(scene):
+    ''' Duplicates vertex instances and makes them real'''
+    unselect_all_objects()
```
Comment (cedricpinson) [inline blender-2.5/exporter/osg/osgdata.py]: what happens if the users selected objects to export ?

**Comment by AurL** [comment]
Ho, right. I didn't think about this case. We need to save the selection and restore it after the preprocess.

File: blender-2.5/exporter/osg/osgdata.py (hunk 1 above)
Comment (AurL) [inline blender-2.5/exporter/osg/osgdata.py]: Ho, right. I didn't think about this case. We need to save the selection and restore it after the preprocess.

**Comment by AurL** [comment]
Update: selection is saved and restored after preprocess functions

**Comment by mrchlblng** [comment]
Good for me

**Pull request merged**